pkg/trace/traceutil/otel_util.go

```diff
@@ -205,8 +205,140 @@ func GetOTelResource(span ptrace.Span, res pcommon.Resource) (resName string) {
-func GetOTelOperationName(
+// GetOTelResourceV2 returns the DD resource name based on OTel span and resource attributes.
+func GetOTelResourceV2(span ptrace.Span, res pcommon.Resource) (resName string) {
+	resName = GetOTelAttrValInResAndSpanAttrs(span, res, false, "resource.name")
+	if resName == "" {
+		if m := GetOTelAttrValInResAndSpanAttrs(span, res, false, "http.request.method", semconv.AttributeHTTPMethod); m != "" {
+			if m == "_OTHER" {
+				m = "HTTP"
+			}
+			// use the HTTP method + route (if available)
+			resName = m
+			if span.Kind() == ptrace.SpanKindServer {
+				if route := GetOTelAttrValInResAndSpanAttrs(span, res, false, semconv.AttributeHTTPRoute); route != "" {
+					resName = resName + " " + route
+				}
+			}
+		} else if m := GetOTelAttrValInResAndSpanAttrs(span, res, false, semconv.AttributeMessagingOperation); m != "" {
+			resName = m
+			// use the messaging operation
+			if dest := GetOTelAttrValInResAndSpanAttrs(span, res, false, semconv.AttributeMessagingDestination, semconv117.AttributeMessagingDestinationName); dest != "" {
+				resName = resName + " " + dest
+			}
+		} else if m := GetOTelAttrValInResAndSpanAttrs(span, res, false, semconv.AttributeRPCMethod); m != "" {
+			resName = m
+			// use the RPC method
+			if svc := GetOTelAttrValInResAndSpanAttrs(span, res, false, semconv.AttributeRPCService); m != "" {
+				// ...and service if available
+				resName = resName + " " + svc
+			}
+		} else {
+			resName = span.Name()
+		}
+	}
+	if len(resName) > MaxResourceLen {
+		resName = resName[:MaxResourceLen]
+	}
+	return
+}
+
+// GetOTelOperationNameV2 returns the DD operation name based on OTel span and resource attributes and given configs.
+func GetOTelOperationNameV2(
+	span ptrace.Span,
+) string {
+	if operationName := GetOTelAttrVal(span.Attributes(), false, "operation.name"); operationName != "" {
+		return operationName
+	}
+
+	isClient := span.Kind() == ptrace.SpanKindClient
+	isServer := span.Kind() == ptrace.SpanKindServer
+
+	// http
+	if method := GetOTelAttrVal(span.Attributes(), false, "http.request.method", semconv.AttributeHTTPMethod); method != "" {
+		if isServer {
+			return "http.server.request"
+		}
+		if isClient {
+			return "http.client.request"
+		}
+	}
+
+	// database
+	if v := GetOTelAttrVal(span.Attributes(), true, semconv.AttributeDBSystem); v != "" && isClient {
+		return v + ".query"
+	}
+
+	// messaging
+	system := GetOTelAttrVal(span.Attributes(), true, semconv.AttributeMessagingSystem)
+	op := GetOTelAttrVal(span.Attributes(), true, semconv.AttributeMessagingOperation)
+	if system != "" && op != "" {
+		switch span.Kind() {
+		case ptrace.SpanKindClient, ptrace.SpanKindServer, ptrace.SpanKindConsumer, ptrace.SpanKindProducer:
+			return system + "." + op
+		}
+	}
+
+	// RPC & AWS
+	rpcValue := GetOTelAttrVal(span.Attributes(), true, semconv.AttributeRPCSystem)
+	isRPC := rpcValue != ""
+	isAws := isRPC && (rpcValue == "aws-api")
+	// AWS client
+	if isAws && isClient {
```

Comment:
do we need to handle aws server ?

Reply:
Could you explain how you think we should handle it? I didn't see it discussed in [the document specifying conventions for operation names](https://docs.google.com/document/d/11YySsQeneNszj_zCIeMCSHTOJHFivrh5olUKRpbVP8A/edit?tab=t.0#heading=h.u36j9t1op6gw)

I could set it to `aws.server.request`, just not sure if ".request" would always be appropriate.